core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -63,7 +64,10 @@ private[log] case class ProducerIdEntry(producerId: Long, producerEpoch: Short,
  * sequence numbers and epochs of each new record. Additionally, this class accumulates transaction metadata
  * as the incoming records are validated.
  */
-private[log] class ProducerAppendInfo(val producerId: Long, initialEntry: ProducerIdEntry, loadingFromLog: Boolean = false) {
+private[log] class ProducerAppendInfo(val producerId: Long,
```

Comment:
It would be useful to document `validateSequenceNumbers` (i.e. why it makes sense for it to be `false`) and `loadingFromLog` (what is the significance). Also, it seems like we have one of these per partition, it may be worth mentioning that.